corehq/sql_db/config.py

```diff
@@ -141,16 +101,58 @@ def get_django_shard_map(self):
         db_shards = self._get_django_shards()
         return {shard.shard_id: shard for shard in db_shards}
 
+    @classmethod
+    def from_settings(cls):
+        assert settings.USE_PARTITIONED_DATABASE
+        return PlProxyConfig.from_dict(settings.PARTITION_DATABASE_CONFIG)
+
+    @classmethod
+    def from_dict(cls, config_dict):
+        if 'groups' in config_dict:
+            # convert old format
+            proxy_db = config_dict['groups']['proxy'][0]
+        else:
+            proxy_db = config_dict['proxy']
+        shard_map = config_dict['shards']
+        host_map = config_dict.get('host_map', {})
+
+        config = PlProxyConfig(proxy_db, shard_map, host_map, _get_shard_count(shard_map.values()))
+        config.validate()
+        return config
+
+    def validate(self):
```

Comment:
should this still be a private method?

Reply:
It could be though I don't thin kit's necessary